src/main/java/mate/jdbc/dao/ManufacturersDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+import mate.jdbc.models.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturersDaoImpl implements ManufacturersDao {
+    @Override
+    public List<Manufacturer> getAll() {
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        String getAllManufacturersRequest = "SELECT * FROM manufacturers WHERE is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllManufacturersStatement
+                         = connection.prepareStatement(getAllManufacturersRequest)) {
+            ResultSet resultSet = getAllManufacturersStatement.executeQuery();
+            if (resultSet == null) {
+                return new ArrayList<>();
+            }
+            while (resultSet.next()) {
+                Manufacturer manufacturer = retrieveData(resultSet);
+                allManufacturers.add(manufacturer);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all manufacturers", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturerRequest
+                = "INSERT INTO manufacturers (name, country) values(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createManufacturerStatement
+                         = connection.prepareStatement(insertManufacturerRequest,
+                        Statement.RETURN_GENERATED_KEYS);
+        ) {
+            createManufacturerStatement.setString(1, manufacturer.getName());
+            createManufacturerStatement.setString(2, manufacturer.getCountry());
+            createManufacturerStatement.executeUpdate();
+            ResultSet generatedKeys = createManufacturerStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create manufacturer" + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getManufacturerRequest = "SELECT * FROM manufacturers "
+                + "WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacturerStatement
+                        = connection.prepareStatement(getManufacturerRequest);) {
+            getManufacturerStatement.setLong(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            if (resultSet == null) {
+                return Optional.empty();
+            }
+            Manufacturer manufacturer = null;
+            while (resultSet.next()) {
```

Comment:
```suggestion
            if (resultSet.next()) {
```
You can't get more than one item here.